pyuvsim/simsetup.py

```diff
@@ -1668,23 +1705,23 @@ def initialize_uvdata_from_params(obs_params, return_beams=None):
         extra_keywords['world'] = tele_params.get('world')
     uvparam_dict['extra_keywords'] = extra_keywords
 
-    # Parse frequency structure
-    freq_dict = param_dict['freq']
-    uvparam_dict.update(parse_frequency_params(freq_dict))
-
     # Parse time structure
     time_dict = param_dict['time']
     uvparam_dict.update(parse_time_params(time_dict))
+    logger.info("Finished Setup of Time Dict")
 
     # There does not seem to be any way to get polarization_array into uvparam_dict, so
     # let's add it explicitly.
-    if "polarization_array" in param_dict:
-        uvparam_dict['polarization_array'] = param_dict['polarization_array']
+    if param_dict.get("polarization_array", None) is not None:
+        uvparam_dict['polarization_array'] = np.array(param_dict['polarization_array'])
 
     # Parse polarizations
     if uvparam_dict.get('polarization_array', None) is None:
+        print("Doing this...")
         uvparam_dict['polarization_array'] = np.array([-5, -6, -7, -8])
+
     if 'Npols' not in uvparam_dict:
+        print("UVparam_dict: pol", uvparam_dict['polarization_array'])
```

Comment:
If you want to keep this please make it part of the logger so it's not alway printed.